web-bundle/src/main/java/com/graphhopper/swl/StableIdEncodedValues.java

```diff
@@ -43,17 +46,39 @@ final String getStableId(boolean reverse, EdgeIteratorState edge) {
         return Long.toUnsignedString(Longs.fromByteArray(stableId));
     }
 
-    final void setStableId(boolean reverse, EdgeIteratorState edge, String stableId) {
-        long stableIdLong = Long.parseUnsignedLong(stableId);
-        byte[] stableIdBytes = Longs.toByteArray(stableIdLong);
-
-        if (stableIdBytes.length != 8)
-            throw new IllegalArgumentException("stable ID must be 8 bytes: " + stableId);
+    public final void setStableId(boolean reverse, EdgeIteratorState edge) {
+        byte[] stableId = calculateStableEdgeId(edge);
+        if (stableId.length != 8)
+            throw new IllegalArgumentException("stable ID must be 8 bytes: " + new String(stableId));
 
         UnsignedIntEncodedValue[] idBytes = reverse ? reverseStableIdEnc : stableIdEnc;
         for (int i=0; i<8; i++) {
-            edge.set(idBytes[i], Byte.toUnsignedInt(stableIdBytes[i]));
+            edge.set(idBytes[i], Byte.toUnsignedInt(stableId[i]));
         }
     }
 
+    private static byte[] calculateStableEdgeId(EdgeIteratorState edge) {
+        // todo: how to get type of street?
```

Comment:
this is the OSM type, right? I can't fully remember what we use. Honestly, we could use our own type if that's easier. I couldn't find a "standard" implementation

Reply:
Yeah, that's what we use right now - it's actually just an int corresponding to the OSM way type I think: https://github.com/replicahq/model/blob/master/mapservice/api/network_link_references.py#L66

I guess, if this information isn't easily accessible here, we just need to make sure that what we choose to hash with is sufficient to guarantee all of edges will have unique hashes...it seems like this specific part of the hash string deals with the case where links of different types have complete overlap in their geometries, so we'll need to accommodate that somehow here using the info available in GH edges

Reply:
OK, I actually messed around some more and figured out how to grab this info from a GH edge (luckily, it's encoded by default). The options are [here](https://github.com/graphhopper/graphhopper/blob/dfdd92cf5e0d7f2c6c2289b98b646dfb6cfd36c1/core/src/main/java/com/graphhopper/routing/ev/RoadClass.java#L25), they seem to line up pretty well with the current enum we're using for road type.